ddtrace/contrib/anthropic/_streaming.py

```diff
@@ -0,0 +1,329 @@
+            )
+            self._dd_span.finish()
+            raise
+        except Exception:
+            self._dd_span.set_exc_info(*sys.exc_info())
+            self._dd_span.finish()
+            raise
+
+    def __stream_text__(self):
+        # this is overridden because it is a helper function that collects all stream content chunks
+        for chunk in self:
+            if chunk.type == "content_block_delta" and chunk.delta.type == "text_delta":
+                yield chunk.delta.text
+
+
+class TracedAnthropicAsyncStream(BaseTracedAnthropicStream):
+    async def __aenter__(self):
+        await self.__wrapped__.__aenter__()
+        return self
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        await self.__wrapped__.__aexit__(exc_type, exc_val, exc_tb)
+
+    def __aiter__(self):
+        return self
+
+    async def __anext__(self):
+        try:
+            chunk = await self.__wrapped__.__anext__()
+            self._streamed_chunks.append(chunk)
+            return chunk
+        except StopAsyncIteration:
+            _process_finished_stream(
+                self._dd_integration,
+                self._dd_span,
+                self._args,
+                self._kwargs,
+                self._streamed_chunks,
+            )
+            self._dd_span.finish()
+            raise
+        except Exception:
+            self._dd_span.set_exc_info(*sys.exc_info())
+            self._dd_span.finish()
+            raise
+
+    async def __stream_text__(self):
+        # this is overridden because it is a helper function that collects all stream content chunks
+        async for chunk in self:
+            if chunk.type == "content_block_delta" and chunk.delta.type == "text_delta":
+                yield chunk.delta.text
+
+
+class TracedAnthropicStreamManager(BaseTracedAnthropicStream):
+    def __enter__(self):
+        stream = self.__wrapped__.__enter__()
+        traced_stream = TracedAnthropicStream(
+            stream,
+            self._dd_integration,
+            self._dd_span,
+            self._args,
+            self._kwargs,
+        )
+        # we need to set a text_stream attribute so we can trace the yielded chunks
+        traced_stream.text_stream = traced_stream.__stream_text__()
+        return traced_stream
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.__wrapped__.__exit__(exc_type, exc_val, exc_tb)
+
+
+class TracedAnthropicAsyncStreamManager(BaseTracedAnthropicStream):
+    async def __aenter__(self):
+        stream = await self.__wrapped__.__aenter__()
+        traced_stream = TracedAnthropicAsyncStream(
+            stream,
+            self._dd_integration,
+            self._dd_span,
+            self._args,
+            self._kwargs,
+        )
+        # we need to set a text_stream attribute so we can trace the yielded chunks
+        traced_stream.text_stream = traced_stream.__stream_text__()
```

Comment:
Can we move this to being set at the `TracedAnthropicAsyncStream.__init__()`? It makes more sense to override at the `TracedAsyncStream` level, not from the `TracedAsyncStreamManager` level